src/tiled/tileseteditor.cpp

```diff
@@ -196,7 +196,13 @@ void TilesetEditor::addDocument(Document *document)
 
     TilesetView *view = new TilesetView(mWidgetStack);
     view->setTilesetDocument(tilesetDocument);
-    view->setZoomable(new Zoomable(this));
+
+    auto zoomable = new Zoomable(this);
+    auto fileName = tilesetDocument->fileName();
+    qreal zoom = Preferences::instance()->TilesetScale(fileName);
```

Comment:
Actually, I think you can move this loading and saving of the zoom level into the `TilesetView`. That way it will work everywhere, including the Tilesets dock in the map editor and the tileset view in the animation editor.

That does mean that the Zoomable instance should be created and owned by the TilesetView. You can remove `TilesetView::setZoomable` and instead only have a getter. You'll need to use `Zoomable::setComboBox` in `TilesetDock::currentTilesetChanged` to make sure the combo box in the TilesetsDock displays the right zoom level.